Content/GUI/ArmorChargeUI.cs

```diff
@@ -0,0 +1,37 @@
+﻿using StarlightRiver.Content.Configs;
+using StarlightRiver.Core.Loaders.UILoading;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Terraria.GameContent.UI.Elements;
+using Terraria.UI;
+
+namespace StarlightRiver.Content.GUI
+{
+	internal class ArmorChargeUI : SmartUIState
+	{
+		private static string message;
+
+		public override bool Visible => ModContent.GetInstance<GUIConfig>().ObviousArmorCharge;
+
+		public override int InsertionIndex(List<GameInterfaceLayer> layers)
+		{
+			return layers.FindIndex(layer => layer.Name.Equals("Vanilla: Mouse Text"));
+		}
+
+		public override void Draw(SpriteBatch spriteBatch)
+		{
+			if (message != "" && ModContent.GetInstance<GUIConfig>().ObviousArmorCharge)
```

Comment:
use !String.IsNullOrEmpty(message) instead of message != "" to avoid errors with undefined strings